lib/ansible/plugins/connection/winrm.py

```diff
@@ -722,7 +723,13 @@ def _wrapper_payload_stream(self, payload: bytes, buffer_size: int = 200000) ->
 
     def exec_command(self, cmd: str, in_data: bytes | None = None, sudoable: bool = True) -> tuple[int, bytes, bytes]:
         super(Connection, self).exec_command(cmd, in_data=in_data, sudoable=sudoable)
-        cmd_parts = self._shell._encode_script(cmd, as_list=True, strict_mode=False, preserve_rc=False)
+
+        encoded_prefix = self._shell._encode_script('', as_list=False, strict_mode=False, preserve_rc=False)
+        if cmd.startswith(encoded_prefix):
+            # Avoid double encoding the script
+            cmd_parts = cmd.split(" ")
+        else:
+            cmd_parts = self._shell._encode_script(cmd, as_list=True, strict_mode=False, preserve_rc=False)
 
         # TODO: display something meaningful here
```

Comment:
lol, that's a really old TODO :laughing:

Reply:
Heh, winrm is a quite old connection plugin now :)